infrastructure/transform/feature-data-store.tf

```diff
@@ -0,0 +1,139 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+resource "random_password" "sql_admin_password" {
+  length  = 16
+  special = true
+}
+
+# Azure SQL logical server, public access disabled - will use private endpoints for access
+resource "azurerm_mssql_server" "sql_server_features" {
+  name                                 = "sql-server-features-${lower(var.naming_suffix)}"
+  location                             = var.core_rg_location
+  resource_group_name                  = var.core_rg_name
+  version                              = "12.0"
+  administrator_login                  = local.sql_server_features_admin_username
+  administrator_login_password         = random_password.sql_admin_password.result
+  public_network_access_enabled        = var.local_mode
+  outbound_network_restriction_enabled = true
+  tags                                 = var.tags
+}
+
+# optional firewall rule when running in local_mode
+resource "azurerm_mssql_firewall_rule" "deployer_ip_exception" {
+  count            = var.local_mode == true ? 1 : 0
+  name             = "DeployerIP"
+  server_id        = azurerm_mssql_server.sql_server_features.id
+  start_ip_address = var.deployer_ip_address
+  end_ip_address   = var.deployer_ip_address
+}
+
+# Enable Transparent Data Encryption (TDE) with a Service Managed Key
+resource "azurerm_mssql_server_transparent_data_encryption" "sql_server_features_encryption" {
+  server_id = azurerm_mssql_server.sql_server_features.id
+}
+
+# Azure SQL database, basic + small for dev
+# TODO: Rightsize for prod -> https://github.com/UCLH-Foundry/FlowEHR/issues/63
+resource "azurerm_mssql_database" "feature_database" {
+  name                 = "sql-db-features"
+  server_id            = azurerm_mssql_server.sql_server_features.id
+  collation            = "SQL_Latin1_General_CP1_CI_AS"
+  license_type         = "LicenseIncluded"
+  max_size_gb          = 2
+  sku_name             = "Basic"
+  storage_account_type = "Local"
+  zone_redundant       = false
+  tags                 = var.tags
+}
+
+# AAD App + SPN for Databricks -> SQL Access
+resource "azuread_application" "flowehr_databricks_sql" {
+  display_name = "FlowEHR-Databricks-SQL-${var.naming_suffix}"
+}
```

Comment:
we need to work out the permissions required to do this for those with no AD access to the shared dev subscription

Reply:
yes wider discussion - but they will ideally get the `Application Developer` role in AAD, which means they can create + manage their own apps, but not touch others.